File: 30days/day10.ts
const convertDecimalToBinary = (number: number) => {
	const binary: string = (number >>> 0).toString(2);
	let count: number = 0;
	let max: number = 0;
	const { length } = binary;
	for (let i = 0; i < length; i++) {
		if (binary[i] === "1") {
			count++;
		} else {
			count = 0;
		}
		max = max > count ? max : count;
	}
	console.log(max);
};

convertDecimalToBinary(125);

File: HackerRank/30days/go/day10.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func convert_dec_to_bin(n int32) {
	count, max := 0, 0

	binary := strconv.FormatInt(int64(n), 2)
	arr_bin := strings.Split(binary, "")
	for _, v := range arr_bin {
		if strings.Compare(v, "1") == 0 {
			count++
		} else {
			count = 0
		}

		if max < count {
			max = count
		}
	}
	fmt.Println(max)
}
func main() {
	convert_dec_to_bin(125)
}
